modules/gapi/src/backends/oak/goakbackend.cpp

```diff
@@ -375,33 +664,69 @@ cv::gimpl::GOAKExecutable::GOAKExecutable(const ade::Graph& g,
                 m_oak_nodes.at(nh).inputs.reserve(op.args.size());
                 m_oak_nodes.at(nh).outputs.reserve(op.outs.size());
 
+                // Handle Copy operations
+                OAKCopyHelper copy_helper;
+                if (op.k.name == "org.opencv.oak.copy") {
+                    if (m_oak_copy_ops.find(nh) != m_oak_copy_ops.end()) { // actual copy
+                        copy_helper = m_oak_copy_ops[nh];
+                        // Also check if this Copy is passthrough at the same time
+                        // Copy has only 1 output
+                        auto copy_out = nh.get()->outNodes();
+                        GAPI_Assert(copy_out.size() == 1);
+                        // FIXME: consider reworking this logic
+                        // Check if there is an OAK island OP after Copy
+                        for (const auto& copy_next_op : copy_out.front().get()->outNodes()) {
+                            if (std::find(nodes.begin(), nodes.end(), copy_next_op) != nodes.end()) {
+                                m_passthrough_copy_nodes.insert(nh);
+                                break;
+                            }
+                        }
+                    } else { // passthrough copy
+                        m_passthrough_copy_nodes.insert(nh);
+                        // Process this kind of Copy in a different way later (linking stage below)
+                        continue;
+                    }
+                }
+
                 std::vector<cv::GArg> in_ctx_args;
                 in_ctx_args.reserve(op.args.size());
                 for (auto &op_arg : op.args) in_ctx_args.push_back(packInArg(op_arg,
-                                                                           m_oak_nodes.at(nh).inputs));
+                                                                             m_oak_nodes.at(nh).inputs));
                 for (auto &&op_out : op.outs) packOutArg(op_out, m_oak_nodes.at(nh).outputs);
                 GAPI_Assert(!m_oak_nodes.at(nh).inputs.empty());
                 GAPI_Assert(!m_oak_nodes.at(nh).outputs.empty());
 
-                GOAKContext ctx(m_pipeline, m_camera_size, in_ctx_args, m_oak_nodes.at(nh).outputs);
-                m_oak_nodes.at(nh).node = u.k.m_put_f(ctx, m_in_queues);
+                if (ConstOAKGraph(m_g).metadata(nh).contains<cv::gimpl::NetworkParams>()) {
+                    GOAKContext ctx(m_pipeline, m_camera_size, m_oak_infer_info[nh],
+                                    in_ctx_args, m_oak_nodes.at(nh).outputs);
+                    m_oak_nodes.at(nh).node = u.k.m_put_f(ctx, m_in_queues);
+                } else {
+                    GOAKContext ctx(m_pipeline, m_camera_size,
+                                    copy_helper, in_ctx_args, m_oak_nodes.at(nh).outputs);
```

Comment:
This `copy_helper` is for copy operation only, maybe it's better to handle this case separately

Reply:
Are you suggesting to create another `GOAKContext`?